tests/ha/await_upgrade_or_update.pm

```diff
@@ -0,0 +1,32 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Wait for all nodes which are allowed to upgrade or update before.
+# Maintainer: Julien Adamek <jadamek@suse.com>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use hacluster;
+use lockapi;
+
+sub run {
+    my $cluster_name = get_cluster_name;
+    my @word         = (get_required_var('UPDATE_TYPE') eq "update") ? ("Update", "updating", "UPDATED") : ("Upgrade", "upgrading", "UPGRADED");
```

Comment:
Looking at the information that is contained in this array, I think there could be a way to simplify it. For example, it seems the string for the barrier could be calculated as `uc(get_required_var('UPDATE_TYPE')) . 'D'`, although as long as there is no barrier named `NODE_UPGRADE_$cluster_name_NODE$num_node` or `NODE_UPDATE_$cluster_name_NODE$num_node` could also be used. Same with the first word in each array, it could be calculated with `ucfirst(lc(get_required_var('UPDATE_TYPE')))` and the middle word with `s/e$/ing/i`. That way you could have variables for action, current action and barrier tag which would be easier to read than `$word[0]` or `$word[1]`, etc.

Reply:
You are right! I didn't think about that at all, thx

Reply:
> You are right! I didn't think about that at all, thx
> However, for the barrier, I would use `get_var` instead of `get_required_var`, otherwise, all MM tests without this variable in the support server will fail.

Yes, on the `barrier_init` better not to use `get_required_var()`.

Reply:
> Looking at the information that is contained in this array, I think there could be a way to simplify it. For example, it seems the string for the barrier could be calculated as `uc(get_required_var('UPDATE_TYPE')) . 'D'`, although as long as there is no barrier named `NODE_UPGRADE_$cluster_name_NODE$num_node` or `NODE_UPDATE_$cluster_name_NODE$num_node` could also be used. Same with the first word in each array, it could be calculated with `ucfirst(lc(get_required_var('UPDATE_TYPE')))` and the middle word with `s/e$/ing/i`. That way you could have variables for action, current action and barrier tag which would be easier to read than `$word[0]` or `$word[1]`, etc.

After discussing this further, it seems having all the tags and messages in the array could in fact be clearer than calculating the values as long as the module remains around 20 lines long (as it currently is). So feel free to leave this with the array.